app/Http/Controllers/ProductController.php

```diff
@@ -31,13 +36,15 @@ public function index(ProductRequest $request)
             });        
             
             if($request->input('paginate') == 'true'){
-                $isPaginate = true;
-                $products = $query_res->paginate();
+                $pageSize = $request->input('pageSize', 10);
+                $pageNum = $request->input('pageNum', 1);
+                $this->isPaginate = true;
+                $products = $query_res->paginate($pageSize, ['*'], 'page', $pageNum);
             } else{
                 $products = $query_res->get();
             }
 
-            return ResponseHelper::success(ProductCollection::collection($products) , 'Data fetched successfully!',200,$isPaginate);
+            return ResponseHelper::success(ProductCollection::collection($products) , 'Data fetched successfully!',200,$this->isPaginate);
 
         }
         catch (Exception $e) {
```

Comment:
Catch more specific exceptions if possible, instead of the general Exception

Reply:
replaced with your helper class responses